src/layers.c

```diff
@@ -318,6 +318,9 @@ new_layer_surface_notify(struct wl_listener *listener, void *data)
 		struct wlr_output *output = wlr_output_layout_output_at(
 			server->output_layout, server->seat.cursor->x,
 			server->seat.cursor->y);
+		if (!output) {
+			/* FIXME: what to do here? */
```

Comment:
I am not sure how to deal with the new layer_surface in particular, should we just send back some wayland error in that case?

Reply:
The scenegraph for sway seems to simply destroy the surface in that case:

https://github.com/swaywm/sway/blob/993deff56993dd5565061a73b6ca2d65cfa6e135/sway/desktop/layer_shell.c#L323-L332

Reply:
Yes, I think this is it. `wlr_layer_surface_v1_destroy()` should sort the associated scene-node